beep/structure.py

```diff
@@ -165,6 +173,24 @@ def get_interpolated_discharge_cycles(self, v_range=None, resolution=1000):
 
         return result
 
+    def get_interpolated_cycles(self, v_range=None, resolution=1000):
+        """
```

Comment:
Don't we want to filter out diagnostic cycles from being interpolated by this method?

Reply:
I could filter to just the regular cycles, although it would add some complexity to the method. I will try and see if I can make it compact.

Reply:
I think I found a relatively clean method for this

Reply:
Nice!